### PR TITLE
Warn when encrypting suspicious whitespace

### PR DESCRIPTION
**Pull request by toothbrush**
# Description

I've been bitten fairly regularly by adding accidental whitespace to strings i'm about to encrypt; see #38 for details.

This PR introduces a simple warning when `plaintext != trim(plaintext)`.

## Approach

@mdub suggested adding a CLI option with silence/warn/error settings for this whitespace check, but IMHO this warn-by-default/no-warn boolean is probably good enough – for my purposes at least!  However, i'd be happy to go back and improve the flexibility of this CLI option if that's what folks want.

## User experience

Let's say someone wants to encrypt the token `foo` using shush.  With my addition, that'll go like this – because remember that `echo "foo"` adds a newline.

```shellsession
$ echo "foo" | shush encrypt alias/buildkite-non-production
shush: 🚨 WARNING: Plaintext contains suspicious whitespace, consider --trim, or silence with --no-warn-whitespace
AQIdeadb33f -- whoops, forgot echo -n
$ echo "foo" | shush encrypt --trim alias/buildkite-non-production
AQIb33fcaf3 -- 👍
```

## Drive-by fixes

I upgraded the libs to latest and squashed a few lint warnings my editor was showing.  Happy to back those out of this PR if those updates are unwanted.

Fixes https://github.com/realestate-com-au/shush/issues/38.

### CODE REVIEW

**Comment by toothbrush** [comment]
I also took the liberty of updating the README a bit:

* I think it's a bit of a head-scratcher suggesting people invoke shush interactively, rather than passing something via stdin?
* I added a Dockerfile trick i recently encountered, which removes the need for the TARGETARCH stuff.  What happens is that Docker Engine grabs the right "shush" container for your machine, subsequently you just grab the shush binary from that.  Might be too "clever" though, happy to revert.

File: README.md [hunk 1]
```diff
@@ -78,12 +78,15 @@ In this example, "shush exec":
...
 "shush exec" works well as an entrypoint for Docker images, e.g.
 
-    # Include "shush" to decode KMS_ENCRYPTED_STUFF
-    ARG TARGETARCH
-    RUN curl -fsSL -o /usr/local/bin/shush \
-        https://github.com/realestate-com-au/shush/releases/download/v1.5.4/shush_linux_${TARGETARCH} \
-     && chmod +x /usr/local/bin/shush
-    ENTRYPOINT ["/usr/local/bin/shush", "exec", "--"]
+```Dockerfile
+FROM realestate/shush:1.5.4 AS shush
```
Comment (toothbrush) [inline README.md:82]:
```suggestion
FROM realestate/shush:1.5.5 AS shush
```

**Comment by toothbrush** [comment]
Hey folks – i'm not sure who to poke for a review, maybe @grantbeattie since you committed to this repo recently? 🙏 

Happy for you to accept/reject/tell me what to improve 😁

**Comment by adrianmace** [comment]
Good morning @toothbrush, 

I'm a current REA employee but do not have the necessary rights on this repository. I'll reach out internally and see that this PR gets looked at.

The change is a good one, it's definitely caught me out at least once too! Thanks for your patience on this.

**Review by grantbeattie — APPROVED** [review]
Looks good!

**Comment by grantbeattie** [comment]
Thanks for the PR. It's great to have this so that people can avoid being bitten by this  "feature" :)

I'll kick out a new release once it's built.

Thanks!

**Comment by toothbrush** [comment]
Hey, thanks for getting back to me, and no problem, i know how it can be with maintainership for "somewhat deprecated" software.

I am happy to volunteer as a maintainer, references available etc. but i understand if that's not possible :).

Thanks again.

**Pull request merged**